kernel/Kconfig

```diff
@@ -422,6 +422,8 @@ menu "Work Queue Options"
 config SYSTEM_WORKQUEUE_STACK_SIZE
 	int "System workqueue stack size"
 	default 4096 if COVERAGE
+	default 2048 if BT_MESH && !BT_RECV_IS_RX_THREAD
```

Comment:
let's not pollute this kernel configuration with subsystem related settings.

Reply:
Removed.

Reply:
Removing this will likely cause issues for some Mesh cases. 

Since BT will now use `SYSTEM_WORKQUEUE_STACK_SIZE` and/or applications using `BT_SETTINGS`, as well as `BT_AUDIO`. 

How do ensure (or at least indicate) that specific BT configurations need increased `SYSTEM_WORKQUEUE_STACK_SIZE` values?

Reply:
If we’re not allowed to touch Kconfig entries outside of the subsystem like @nashif says (which does make sense) I think the other option is to use build asserts?

Reply:
> If we’re not allowed to touch Kconfig entries outside of the subsystem like @nashif says (which does make sense) I think the other option is to use build asserts?

If we go from a suggested default (like here) to a specific and hard requirement like BUILD_ASSERT, then we need to spend some additional time on calculating the stack usage, as we probably shouldn't require e.g. 2048 octets for the stack if 1200 is sufficient. 

@Casper-Bonde-Bose You mentioned that native_posix stack sizes were not affected by this config. Does that apply to babblesim as well (ping @aescolar )? In other words, does this value affect the babblesim test cases?

Reply:
@Thalley babblesim use `Controller+Host`, which `CONFIG_BT_RECV_IS_RX_THREAD` selected by https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/controller/Kconfig#L104, so this change not affect babblesim test cases.